converter/src/test/java/org/dswarm/converter/schema/test/SolrXMLDataSourceConfigGeneratorTest.java

```diff
@@ -35,10 +54,26 @@ public void testSolrXMLDataSourceConfigGenerator() throws IOException, XMLStream
 
 		final Schema fincSolrSchema = schemaService.getObject(SchemaUtils.FINC_SOLR_SCHEMA_UUID);
 
-		final String dswarmRoot = GuicedTest.injector.getInstance(Key.get(String.class, Names.named("dswarm.paths.root")));
+		final String dswarmRoot = System.getProperty("user.dir");
 		final String sep = File.separator;
 		final String fileName = "data-config.xml";
-		final String fullFileName = dswarmRoot + sep + "converter" + sep + "target" + sep + "classes" + sep + fileName;
+
+		final StringBuilder sb = new StringBuilder();
+
+		sb.append(dswarmRoot).append(sep);
+
+		final String converterModule = "converter";
+
+		if (!dswarmRoot.endsWith(converterModule)) {
+
+			sb.append(converterModule).append(sep);
+		}
+
+		sb.append("target").append(sep).append("classes").append(sep).append(fileName);
+
+		final String fullFileName = sb.toString();
+
+		LOG.debug("Solr XML data source config file = '{}'", fullFileName);
```

Comment:
This seems like a very, well, interesting way to obtain a resource. @zazi Can you comment on why this is done this way?